packages/build-tools/src/steps/functions/download_artifact.ts

```diff
@@ -0,0 +1,65 @@
+import {
+  BuildFunction,
+  BuildStepInput,
+  BuildStepInputValueTypeName,
+  BuildStepOutput,
+} from '@expo/steps';
+
+export function createDownloadArtifactFunction(): BuildFunction {
+  return new BuildFunction({
+    namespace: 'eas',
+    id: 'download_artifact',
+    name: 'Download artifact',
+    inputProviders: [
+      BuildStepInput.createProvider({
+        id: 'application_archive_url',
```

Comment:
I think it would be more interesting to use a `build_id` here. `eas/download_artifact` needs to allow us to get rid of `getShortApplicationArchiveUrl()` ([Linear task](https://linear.app/expo/issue/ENG-14849/replace-use-of-turtlebuildutilsgetshortapplicationarchiveurl-with)).

Actually, maybe this should be called `eas/download_build` instead?

This may require adding a new endpoint to some `www` router.